### PR TITLE
fix issue when checking file permission and generate key under system context

### PR DESCRIPTION
**Pull request by bingbing8**
https://github.com/PowerShell/Win32-OpenSSH/issues/736
https://github.com/PowerShell/Win32-OpenSSH/issues/735

### CODE REVIEW

File: contrib/win32/win32compat/fileio.c [hunk 1]
```diff
@@ -359,14 +359,21 @@ createFile_flags_setup(int flags, u_short mode, struct createFile_flags* cf_flag
...
 	if ((sid_utf16 = utf8_to_utf16(pwd->pw_sid)) == NULL) {
 		debug3("Failed to get utf16 of the sid string");
 		errno = ENOMEM;
 		goto cleanup;
 	}
-	if ((mode & S_IRWXU) != 0) {
+
+	if (ConvertStringSidToSid(pwd->pw_sid, &owner_sid) == FALSE ||
+		(IsValidSid(owner_sid) == FALSE)) {
+		debug3("failed to retrieve the sid of the pwd");
```
Comment (manojampalam) [inline contrib/win32/win32compat/fileio.c]: pwd doesn't make sense to end user :). say
debug3("cannot retrieve SID of user %s", pwd->name)

File: contrib/win32/win32compat/w32-sshfileperm.c [hunk 2]
```diff
@@ -42,11 +42,10 @@
 
 /*
 * The function is to check if user prepresented by pw is secure to access to the file. 
-* Check the owner of the file is one of these types: Local Administrators groups, system account
+* Check the owner of the file is one of these types: Local Administrators groups, system account, pwd account
```
Comment (manojampalam) [inline contrib/win32/win32compat/w32-sshfileperm.c]: "pwd" here but "pw" on line 44?

Comment (manojampalam) [reply contrib/win32/win32compat/w32-sshfileperm.c]: Overall, we shouldn't be calling out "pwd" - what does it mean after all?

File: contrib/win32/win32compat/fileio.c (hunk 1 above)
Comment (bingbing8) [reply contrib/win32/win32compat/fileio.c]: fixed

**Comment by bingbing8** [comment]
@manojampalam, I updated the trace and comments. I replace pwd user as current user account. please let me know if you have better name.

**Pull request merged**